File: src/main/java/ast/ASTArrayLiteralExpr.java
package ast;

import java.util.List;

public class ASTArrayLiteralExpr extends ASTLiteralExpr {
    List<ASTLiteralExpr> lits;

    public ASTArrayLiteralExpr(ASTFileLocation loc, List<ASTLiteralExpr> lits){
        super(loc, null);
        this.lits = lits;
    }

    public void print(int i){
        printIndent(i);
        System.out.println("ASTArrayLiteralExpr:");
        for(var n = 0; n < lits.size(); n++){
            lits.get(n).print(i+1);
        }
    }
}

File: src/main/kotlin/ast/ASTArrayLiteralExpr.kt
package ast

class ASTArrayLiteralExpr(loc: ASTFileLocation, val lits: List<ASTExpr>) : ASTLiteralExpr(loc, null) {

    override fun print(i: Int) {
        printIndent(i)
        System.out.println("ASTArrayLiteralExpr:")
        for (n in 0 until lits.size) {
            lits[n].print(i + 1)
        }
    }
}